### ISSUE TITLE
Compilation issues on Alpine Linux

### ISSUE THREAD

**Issue by PercussiveElbow** [arcage/net_sample.cr#6]
**Issue**
I've been having issues when using this library during Alpine builds.

When I run `crystal build`  within the Crystal Alpine container the compiler complains that `__in6_u` doesn't exist.  This occurs on this line https://github.com/arcage/net_sample.cr/blob/4ce05c1aba25414d7f7dffda168c8732a1589504/src/net_sample/nic/linux.cr#L53

**Error: undefined method '__in6_u' for LibC::In6Addr**
```
In src/test.cr:4:21

 4 | puts(NetSample::NIC.ifnames)
                         ^------
Error: instantiating 'NetSample::NIC.class#ifnames()'


In lib/net_sample/src/net_sample/nic.cr:21:10

 21 | self.nics.keys
           ^---
Error: instantiating 'NetSample::NIC.class#nics()'


In lib/net_sample/src/net_sample/nic.cr:17:16

 17 | @@nics ||= get_nic_info
                 ^-----------
Error: instantiating 'get_nic_info()'


In lib/net_sample/src/net_sample/nic/linux.cr:53:33

 53 | addr6 = ina.sin6_addr.__in6_u.__u6_addr8
                            ^------
Error: undefined method '__in6_u' for LibC::In6Addr

LibC::In6Addr trace:

  lib/net_sample/src/net_sample/nic/linux.cr:53

              addr6 = ina.sin6_addr.__in6_u.__u6_addr8
```

This library does work outside of Alpine based builds, but this does prevent using this shard in static builds given Alpine is still the only official way to statically compile Crystal programs.

**Steps to replicate**
* Start a container using the official `crystallang/crystal:latest-alpine`  image.
* Run `crystal build` while referencing this library within a container 

I've attached a sample folder that demonstrates this issue.
[net-sample-alpine-issue.tar.gz](https://github.com/arcage/net_sample.cr/files/5302345/net-sample-alpine-issue.tar.gz)

I'm not really sure why exactly this issue occurs but I assume it's something to do with the use of musl vs gnulibc

Thanks for your work on this library!

**Comment by arcage** [comment]
Thank you for reporting.

And, I'm sorry this shard works only with gnulibc in the case of linux.

It's caused by the different definitions of some libc structs(unions) between gnu and musl.

I will try to support muls environments when i have time to do that.

Until that, it may work(sorry, untested) by changing that line to:

addr6 = ina.sin6_addr. **__in6_union.__s6_addr**

**Comment by arcage** [comment]
Hi,

I released v0.2.5 that support musl environment(tested on official alpine container of crystal 0.35.1)

**Comment by PercussiveElbow** [comment]
Can confirm that 0.2.5 fixed compiling on Alpine for me. Many thanks for the quick patch!
